content/docs/development/health-rules.md

```diff
@@ -0,0 +1,168 @@
+  condition as "Unknown" for the reason "OutputNotAvailable". The workload will report the resource's Ready condition as
+  "Unknown" for the reason "MissingValueAtPath".
+
+Otherwise, the object's Healthy condition will be "True". By defining one of several types of health rules, a template
+author can better define the conditions that report healthy or unhealthy status. This will ensure that the proper status
+is returned to users.
+
+## Always Healthy
+
+Some resources simply have no sense of "healthiness" and are meant to be considered always healthy. For instance, a
+ConfigMap would never be unhealthy as it always simply reflects the data values that were last submitted.
+
+Template authors may choose not specify a healthrule at all (i.e., rely on the default of being always healthy). But it
+is clearer to be explicit:
+
+```yaml
+apiVersion: carto.run/v1alpha1
+kind: Cluster[Config|Deployment|Image|Source]Template
+spec:
+  healthRule:
+    alwaysHealthy: {}
+```
+
+This leads to the following status on the [owner object](architecture.md/#owners):
+
+```yaml
+status:
+  conditions:
+    - type: ResourcesHealthy  # <=== aggregates status.resources[*].conditions[type == Healthy].status
+      status: "True"
+      reason: HealthyConditionRule
+    ...
+  resources:
+    - conditions:
+        - type: Healthy
+          status: "True"
+          reason: AlwaysHealthy
+        ...
+```
+
+## Single Condition
+
+This type of health rule is useful for the majority of resources out there as most custom resources implement the
+pattern of providing a condition set under `status.conditions`.
+
+With `singleConditionType` all you need to provide is the type of the condition to look up when evaluating healthiness.
+
+e.g., consider the `kpack/Image` object:
+
+```yaml
+status:
+  conditions:
+    - status: "True"
+      type: Ready
+    - status: "True"
+      type: BuilderReady
+```
+
+Given that it makes use of the `status.conditions` pattern (and `Ready: True` indicates healthiness), we can leverage
+`singleConditionType`:
+
+<!-- prettier-ignore-start -->
+```yaml
+  healthRule:
+    singleConditionType: Ready
+```
+<!-- prettier-ignore-end -->
+
+Cartographer will then consider the named condition (in this case `Ready`) and evaluate healthiness as the status on
+that condition:
+
+- healthy: status == true
+- unhealthy: status == false
+- unknown: anything else
+
+The `message` field of the specified condition will be replicated on the owner object.
+
+## Multi Match
+
+With multiMatch we're able to specify more than one matching rule for determining healthiness. This is the most flexible
+of the health rules.
+
+For some controllers, two conditions must be met for the object to be healthy. For example in Deployments, users want
+both the `Available` and `Progressing` conditions to be true. For other resources, one condition indicates the object is
```

Comment:
nit (not super sure as a non-native speaker though)

```suggestion
both the `Available` and `Progressing` conditions to be true. For other resources, one condition indicates that the object is
```

Reply:
I am not enough of a grammar nerd to follow the longer answer in this discussion, but the consensus is "that" is optional.

https://english.stackexchange.com/questions/68379/which-is-correct-indicates-or-indicates-that